### ISSUE TITLE
Addition of NumPy like methods in Linear Algebra module.

### ISSUE THREAD

**Issue by prashanth018** [chapel-lang/chapel#18091]
### Summary of Problem:
As part of [this](https://github.com/chapel-lang/chapel/pull/18042) feature (implementation of onenormestimate for Matrix Exponentials), I have implemented few methods which could also be used publicly. Nevertheless, these methods are open for improvements (performance/readability). Also please do comment on what other generic numpy-like methods could be added to LA. Let this issue be used as tracking for suggestions or other discussions. 

### Methods added so far:

```
sparseDenseMatmul: Mutliplies Sparse and Dense Matrix

sign_round_up: Rounds up the elements of the Matrix to 1.0 if element >= 0, -1.0 otherwise

argsort: Given an array A, returns indices corresponding to the sorted ordering of A.

elementary_vector: Returns an index's unit vector of size n.

zeros: Given the Domain and element-type, the method returns a Matrix of zeros.

ones: Given the Domain and element-type, the method returns a Matrix of ones.

max: Gets max along specified axes. (Method for Sparse matrix) [currently only supports axes 0 and 1]

absSum: Gets sums along specified axes. (Added Methods for both Sparse and Dense matrix) [currently only supports axes 0 and 1]
```

**Comment by e-kayrakli** [comment]
@prashanth018 -- To me the main question at this stage is whether we want the functions you've listed above to be user-facing or not. They're inspired by numpy and was necessary for you to implement the one-norm-estimate thingy in the LinearAlgebra module. It could also help the discussion if you can show the full interface that you have in mind for each of these functions.

Some specific thoughts:

- sparseDenseMatmul: I think we can incorporate this as an internal helper and call it from our `dot`
- argsort: See: https://github.com/chapel-lang/chapel/issues/5724#issuecomment-488350400, ping @ronawho and @mppf 
- zeros/ones: I am not against. However, these shouldn't have the type argument an argument with a default value as in your PR, if we make them user facing. Then a line like `var S = zeros(wD);` would become `var S = zeros(wD, real);` and I am not entirely sure how much of an improvement it is over `var S: [wD] real = 0.0;`. I can see them being useful in creating a matrix and passing it over to a function directly without creating a temporary variable, though. Even then, you can probably pass `[wD] 0.0`. That being said I don't see myself as a LinearAlgebra user to assess the need for them.
- max: When/if we have partial reductions, I think this can be easily represented with a partial reduction. Until then, we can add a user-facing function, I guess. We must have a `min` function, as well if we're adding this.
- absSum: Does something like this exist in numpy? I think you should edit your OP to specify that this actually gets sums of absolute values over an axis. I think this is similar to `max` that it should be handled by a partial reduction when we have it.
- sign_round_up, elementary_vector: No comment, besides that their names should be camelCase.

**Comment by lydia-duncan** [comment]
I agree with Engin about showing more of the interface and using camelCase.  Otherwise, these seem generally useful though I do wonder if there's a way to give `sparseDenseMatmul` a shorter name.

**Comment by e-kayrakli** [comment]
> I do wonder if there's a way to give sparseDenseMatmul a shorter name.

I hope that function is not going to be user-facing, at all. It'll be an internal helper that `LinearAlgebra.dot` calls if the operands are sparse and dense.
